### PR TITLE
Add timestamp (wall time) information on the Event 

### PR DESCRIPTION
**Pull request by marcingrzejszczak**
fixes gh-4032

### CODE REVIEW

File: micrometer-observation/src/main/java/io/micrometer/observation/Observation.java [hunk 1]
```diff
@@ -1282,9 +1285,62 @@ public String getContextualName() {
...
                 }
 
+                @Override
+                public long getTimestamp() {
+                    return this.timestamp;
+                }
+
+                @Override
+                public TimeUnit getTimestampTimeUnit() {
+                    return TimeUnit.MILLISECONDS;
```
Comment (jonatan-ivanov) [inline micrometer-observation/src/main/java/io/micrometer/observation/Observation.java:1295]: Is this needed? Can we just say in the javadocs that it is the current wall time in ms?

**Review by jonatan-ivanov — COMMENTED** [review]
I'm not sure I understand the use-case (I made a [comment on the issue](https://github.com/micrometer-metrics/micrometer/issues/4032#issuecomment-1682748138)).

File: micrometer-observation/src/main/java/io/micrometer/observation/Observation.java (hunk 1 above)
Comment (marcingrzejszczak) [reply micrometer-observation/src/main/java/io/micrometer/observation/Observation.java:1295]: That's what I did originally. I even named the method `getTimestampInMs` or sth like that. With 2 methods there is no doubt about what the units are and you are able to modify it as you please. Without that you can make easy mistakes (like I did when I've forgotten that timestamp in Brave is micro not milliseconds :shrug: ). WDYT @shakuzen ?

**Comment by jonatan-ivanov** [comment]
The things I changed in the PR:
- Rebased it on `main`.
- Moved the anonymous `Event` classes out to a `SimpleEvent` class to avoid code duplication (and anonymous classes).
- I consolidated the interface changes into a single `getWallTime` method. Please let me know what you think about this, we can always add a `getWallTimeUnit` as an extra if it will be confusing (tried to document it to make it more clear).

**Pull request merged**